lib/match.js

```diff
@@ -3,14 +3,15 @@
 var safeEqual = require('./safeEqual')
 
 function equal (a, b) {
-  var result = false
   if (typeof a !== 'object') {
-    result = safeEqual(a, b)
+    return safeEqual(a, b)
   } else if (a instanceof RegExp) {
-    result = a.test(b)
+    if (b instanceof RegExp) {
+      return b.toString() === b.toString()
```

Comment:
I think comparing `regex.source` is better.

Reply:
agree